### DISCUSSION TITLE
How to make Cli or JS Api work with windi config file

### DISCUSSION THREAD

**Discussion by SohanChotia** [windicss/windicss#427] [Q&A]
I m not much familiar with all these build tools and not much fan of installing so any npm libs related to build tools for simple html based projects.

I have used tailwind and I prefer to install npm libs globally and be able to use them in any projects.
Tailwind works with all its configs and JIT with postcss and I can make it work with one npm command.

`set \"NODE_ENV=production\" && postcss -w ./resources/css/app.css -o ./public/css/app.css`

I really liked 2 features of windi which Tailwind doesn't have
- attribute mode
- option to parse custom style blocks in any source file( love it ), 
   instead of forcing to write all quick styles needed for one custom page in global style.css file

So the question is how can I make it work with either 
- CLI
- JS API

I tried cli with simple config
`windicss *.php -to public/css/app1.css --config windi.config.js`

```
module.exports = {
  extract: {  // accepts globs and file paths relative to project root
   include: [
      'resources/views/**/*.blade.php',
    ],
    exclude: [   "vendor/**/*", "node_modules/**/*", ".git/**/*"  ]
  },
  darkMode: 'class', // or 'media'
  plugins: [
    require('windicss/plugin/filters'),
    require('windicss/plugin/forms'),
    require('windicss/plugin/aspect-ratio'),
    require('windicss/plugin/line-clamp'),
    require('windicss/plugin/typography')({
      modifiers: ['DEFAULT', 'sm', 'lg', 'red'],
    }),
  ],
}
```
It runs but keeps running and says 
**Allocation failed - JavaScript heap out of memory**

So I need to know, how (if possible) to make it work with either cli or js api with all config options and extraction working

`windicss *.php -to public/css/app1.css --config windi.config.js`
OR

`node windi.js`
and using some code in js api

thnx

**Comment by IgnisDa** [comment]
Windicss works with traditional html files, so there might be a problem with your setup. Can you post more info? O.S. node version and other relevant info?
